### ISSUE TITLE
WiFi

### ISSUE THREAD

**Issue by Ionitron** [capacitor-community/proposals#9]
## Plugin Request

**Name**: WiFi
**Package**: `@capacitor-community/wifi`

### Platform(s)

Android, iOS

### Existing Solutions

https://github.com/sushichop/cordova-plugin-wifi-manager

### Description

- Connecting to a network
- Disconnecting from a network
- Listing available networks

**Comment by Ionitron** [comment]
Original request: https://github.com/ionic-team/capacitor/issues/1897

**Comment by AE1NS** [comment]
What about this?
https://github.com/ionic-team/ionic-native/tree/master/src/%40ionic-native/plugins/wifi-wizard-2

**Comment by imhoffd** [comment]
@AE1NS I added it to the list of existing solutions, but these proposals are for Capacitor plugins, not Cordova plugins.

**Comment by AE1NS** [comment]
Next one:
https://github.com/digaus/capacitor-wifi
Or better this fork with some more updates:
https://github.com/HalleyAssist/capacitor-wifi

Install:
`npm i capacitor-wifi@git://github.com/HalleyAssist/capacitor-wifi.git`

Update MainActivity.java in Android:
```
import com.digaus.capwifi.Wifi; <-- add

this.init(savedInstanceState, new ArrayList<Class<? extends Plugin>>() {{
    ...
    add(Wifi.class); <-- add
}});
```

Usage:
```
import { Plugins } from '@capacitor/core';
const { Wifi } = Plugins;
...
// i.e.
Wifi.connect({ ssid: 'My wifi network ssid' }).then(() => {
    console.log('connected');
});
```

**Comment by imhoffd** [comment]
@digaus Any interest in joining the [Capacitor community](https://github.com/capacitor-community/welcome) with your Wi-Fi plugin?

**Comment by digaus** [comment]
> @digaus Any interest in joining the [Capacitor community](https://github.com/capacitor-community/welcome) with your Wi-Fi plugin?

It's basically copy paste from WifiWizard2

Also does not work sometimes on all Android version. Currently have no time and resources to look into that. :/

**Comment by AE1NS** [comment]
@HalleyAssist you made some more commits in your fork, maybe its more stable?

**Comment by imhoffd** [comment]
WifiWizard2 is Apache-2.0 licensed, which makes it difficult to create legal derivative works (I'm not a lawyer). The Capacitor community generally prefers the MIT license which is much more permissive.

**Comment by digaus** [comment]
Hey everyone.

I am about to get back to my Wifi Plugin and do a rewrite with the newest version of capacitor. 

Unfortunately I do only own the latest iOS / Android Versions. Is there someone willing to implement and test the parts for the older versions?

**Comment by digaus** [comment]
Have started implementing:

https://github.com/digaus/community-capacitor-wifi

Already done are `getIP()` and `getSSID()` for iOS, Android and Electron (not 100% tested)

**Comment by sultanmyrza** [comment]
Still working on this project?

**Comment by digaus** [comment]
> Still working on this project?

Yes but no official releases so far. It should work as is. Recently fixed some issues with iOS.

Just run npm install on my master and it should work.

**Comment by riderx** [comment]
You can add your plugin to [awesome-capacitor](https://github.com/riderx/awesome-capacitor) if that is useful please take time to star the repo :)

**Comment by sultanmyrza** [comment]
@digaus plugin https://github.com/digaus/community-capacitor-wifi worked for our [case](https://github.com/numbersprotocol/capture-lite).

**Comment by michaelpeterlee** [comment]
@codext/capacitor-wifi
